crates/paimon/src/error.rs

```diff
@@ -38,3 +38,12 @@ pub enum Error {
         source: opendal::Error,
     },
 }
+
+#[derive(Debug, Snafu)]
+pub enum DataTypeError {
```

Comment:
Hi, we usually expose only one error in each crate to keep them user-friendly. I will suggest to add `DataTypeInvalid` as a new enum in `Error` directly. In this way, we can use `paimon::Error` everywhere.